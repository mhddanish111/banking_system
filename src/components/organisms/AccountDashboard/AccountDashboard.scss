@import "../../../styles/mixins.scss";
@import "../../../styles/spacing.scss";
.account-dashboard-container {
  .account-container {
    @include tablet-and-over {
      display: flex;
    }
  }
  .card-section {
      margin-top: $space-l;
  }
}
