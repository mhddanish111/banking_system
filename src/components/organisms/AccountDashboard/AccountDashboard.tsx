import React, { useEffect } from 'react';
import { isEmpty } from 'lodash';
import Heading from '../../atoms/Heading';
import AccountTiles from '../../molecules/AccountTiles';
import CardTiles from '../../molecules/CardTiles';
import { AccountData } from '../../../type/AccountType';
import GetContext from '../../../context/GetContext';
import { addAccounts, addCards } from '../../../context/actions';
import './AccountDashboard.scss';

const AccountDashboard: React.FC<AccountData> = (props: AccountData) => {
  const { accountTypeData = [], creditCardData = [] } = props;
  const { dispatch, state = {} }: any = GetContext();
  const { accounts = [], cards = [] } = state;
  useEffect(() => {
    if (isEmpty(accounts) && accountTypeData?.length > 0) {
      dispatch(addAccounts(accountTypeData));
    }
    if (isEmpty(cards) && creditCardData?.length > 0) {
      dispatch(addCards(creditCardData));
    }
  }, [accounts, cards, accountTypeData, creditCardData, dispatch]);
  return (
    <div className="row account-dashboard-container">
      {!isEmpty(accountTypeData) && (
        <section id="account-section" className="col-xs-12">
          <Heading
            HeadingType="h2"
            className="col-xs-12"
            ariaLabel="account heading"
          >
            Accounts
          </Heading>
          <div className="col-xs-12 account-container">
            {accountTypeData.map((accounts, index) => {
              const key = `account-tiles-${index}`;
              return <AccountTiles accounts={accounts} key={key} />;
            })}
          </div>
        </section>
      )}

      {!isEmpty(creditCardData) && (
        <section id="account-section" className="col-xs-12 card-section">
          <Heading
            HeadingType="h2"
            className="col-xs-12"
            ariaLabel="account heading"
          >
            Credit Cards
          </Heading>
          <div className="col-xs-12 account-container">
            {creditCardData.map((cards, index) => {
              const key = `account-tiles-${index}`;
              return <CardTiles cards={cards} key={key} />;
            })}
          </div>
        </section>
      )}
    </div>
  );
};

export default AccountDashboard;
