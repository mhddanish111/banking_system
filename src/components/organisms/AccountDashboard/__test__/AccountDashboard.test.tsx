import { shallow } from 'enzyme';
import AccountDashboard from '../AccountDashboard';

describe('<AccountDashboard />', () => {
  const accounts = {
    _id: '63c1519e9919f0f41a86bd94',
    type: 'Accounts',
    accountType: 'Savings Account',
    categoryName: 'Regular Account',
    accountNumber: 123456789,
    amount: 10000,
    debitCardCSVNumber: 123,
    debitCardNumber: 12312313123,
    expiryMonth: 1,
    expiryYear: 2023,
    limit: 0,
    maxAmount: 0,
    minAmount: 0,
    charge: 0,
    users: [
      {
        _id: '63c1519e9919f0f41a86bd94',
        firstName: 'test',
        lastName: 'test',
      },
    ],
  };
  const cards = {
    type: 'Accounts',
    accountType: 'Savings Account',
    categoryName: 'Regular Account',
    accountNumber: 123456789,
    amount: 10000,
    debitCardCSVNumber: 123,
    debitCardNumber: 12312313123,
    expiryMonth: 1,
    expiryYear: 2023,
    limit: 0,
    maxAmount: 0,
    minAmount: 0,
    _id: '12345',
    charge: 0,
    users: [
      {
        _id: '12345',
        firstName: 'test',
        lastName: 'test',
      },
    ],
  };
  const accountTypeData = [accounts];
  const creditCardData = [cards];

  const accountData = [accounts, cards];

  const wrapped = shallow(
    <AccountDashboard
      accountData={accountData}
      accountTypeData={accountTypeData}
      creditCardData={creditCardData}
    />,
  );
  it('should render the Title Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
