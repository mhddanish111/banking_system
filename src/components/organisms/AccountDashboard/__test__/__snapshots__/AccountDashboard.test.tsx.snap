// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountDashboard /> should render the Title Component correctly 1`] = `
<div
  className="row account-dashboard-container"
>
  <section
    className="col-xs-12"
    id="account-section"
  >
    <Heading
      HeadingType="h2"
      ariaLabel="account heading"
      className="col-xs-12"
    >
      Accounts
    </Heading>
    <div
      className="col-xs-12 account-container"
    >
      <AccountTiles
        accounts={
          Object {
            "_id": "63c1519e9919f0f41a86bd94",
            "accountNumber": 123456789,
            "accountType": "Savings Account",
            "amount": 10000,
            "categoryName": "Regular Account",
            "charge": 0,
            "debitCardCSVNumber": 123,
            "debitCardNumber": 12312313123,
            "expiryMonth": 1,
            "expiryYear": 2023,
            "limit": 0,
            "maxAmount": 0,
            "minAmount": 0,
            "type": "Accounts",
            "users": Array [
              Object {
                "_id": "63c1519e9919f0f41a86bd94",
                "firstName": "test",
                "lastName": "test",
              },
            ],
          }
        }
        key="account-tiles-0"
      />
    </div>
  </section>
  <section
    className="col-xs-12 card-section"
    id="account-section"
  >
    <Heading
      HeadingType="h2"
      ariaLabel="account heading"
      className="col-xs-12"
    >
      Credit Cards
    </Heading>
    <div
      className="col-xs-12 account-container"
    >
      <CardTiles
        cards={
          Object {
            "_id": "12345",
            "accountNumber": 123456789,
            "accountType": "Savings Account",
            "amount": 10000,
            "categoryName": "Regular Account",
            "charge": 0,
            "debitCardCSVNumber": 123,
            "debitCardNumber": 12312313123,
            "expiryMonth": 1,
            "expiryYear": 2023,
            "limit": 0,
            "maxAmount": 0,
            "minAmount": 0,
            "type": "Accounts",
            "users": Array [
              Object {
                "_id": "12345",
                "firstName": "test",
                "lastName": "test",
              },
            ],
          }
        }
        key="account-tiles-0"
      />
    </div>
  </section>
</div>
`;
